pypesto/optimize/util.py

```diff
@@ -11,31 +13,59 @@ def check_hdf5_mp(history_options: HistoryOptions,
     """
     This function checks whether we use another engine than
     `SingleCoreEngine` and whether we use a `Hdf5History`. If
-    that is the case, i rephrases the filename to create a hdf5 file
+    that is the case, it rephrases the filename to create an hdf5 file
     per start. Returns filename for later usage in filling hdf5 file
     otherwise return None.
+
+    Paramters
+    ---------
+    engine:
+        The Engine which is used in the optimization
+    history_options:
+        The HistoryOptions used in the optimization
+
+    Returns
+    -------
+    filename:
+        string containing the original filename.
     """
     if not isinstance(engine, SingleCoreEngine):
         filename = history_options.storage_file
-        fn, type = os.path.splitext(history_options.storage_file)
-        # creates a folder in which the hdf5 files will be saved
-        if not os.path.exists(fn):
-            os.mkdir(fn)
-        fn = fn + '/' + fn + '_{id}'
+        file_path = Path(filename)
+        fn = file_path.parent / (file_path.stem)
+        # create directory with same name as original file
+        fn.mkdir(parents=True, exist_ok=True)
+        fn = fn / (file_path.stem + '_{id}' + file_path.suffix)
+        fn = str(fn)
         history_options.storage_file = fn + type
```

Comment:
```suggestion
        # create directory with same name as original file stem
        partial_file_path = (
            file_path.parent / file_path.stem /
            (file_path.stem + '_{id}' + file_path.suffix)
        )
        partial_file_path.mkdir(parents=True, exist_ok=True)
        history_options.storage_file = str(partial_file_path)
```

No need to do `mkdir` before including the filename in the path.

Also I guess the `+ type` is a typo

Reply:
oh yes that is a typo. But I tested it and when I had a path like `a/b/c.txt` it created a folder called `c.txt` that's why that change was included

Reply:
Ah true. Fixed in the suggestion (can use `Path.parent.mkdir` instead), if you prefer this.

Reply:
Used the suggestion 👍